client/qiskit_serverless/core/client.py

```diff
@@ -182,70 +66,39 @@ def __init__(  # pylint:  disable=too-many-positional-arguments
             name: name of client
             host: host of client a.k.a managers host
             token: authentication token for manager
-            compute_resource: selected compute_resource from provider
-            available_compute_resources: available clusters in provider
         """
         self.name = name
         self.host = host
         self.token = token
-        self.compute_resource = compute_resource
-        if available_compute_resources is None:
-            if compute_resource is not None:
-                available_compute_resources = [compute_resource]
-            else:
-                available_compute_resources = []
-        self.available_compute_resources = available_compute_resources
 
+    # ?¿¿?¿?
     @classmethod
     def from_dict(cls, dictionary: dict):
-        return BaseProvider(**dictionary)
-
-    def job_client(self):
-        """Return job client for configured compute resource of provider.
-
-        Returns:
-            job client
-        """
-        return self.compute_resource.job_client()
-
-    def context(self, **kwargs):
-        """Allocated context for selected compute_resource for provider."""
-        if self.compute_resource is None:
-            raise QiskitServerlessException(
-                f"ComputeResource was not selected for provider {self.name}"
-            )
-        return self.compute_resource.context(**kwargs)
+        return BaseClient(**dictionary)
 
     def __eq__(self, other):
-        if isinstance(other, BaseProvider):
+        if isinstance(other, BaseClient):
             return self.name == other.name
 
         return False
 
     def __repr__(self):
         return f"<{self.name}>"
 
-    def get_compute_resources(self) -> List[ComputeResource]:
-        """Return compute resources for provider."""
-        raise NotImplementedError
-
-    def create_compute_resource(self, resource) -> int:
-        """Create compute resource for provider."""
-        raise NotImplementedError
-
-    def delete_compute_resource(self, resource) -> int:
-        """Delete compute resource for provider."""
-        raise NotImplementedError
-
+    ####################
+    ####### JOBS #######
+    ####################
+    @abstractmethod
     def get_jobs(self, **kwargs) -> List[Job]:
         """Return list of jobs.
 
         Returns:
             list of jobs.
         """
-        raise NotImplementedError
+        pass
 
-    def get_job_by_id(self, job_id: str) -> Optional[Job]:
+    @abstractmethod
+    def get_job(self, job_id: str) -> Optional[Job]:
```

Comment:
Here we are introducing a breaking change. What we usually do is maintain `get_job_by_id` with a `DeprecateWarning` and an internal call to `get_job`.